### PR TITLE
statement-store: Add wire protocol V1 encode/decode edge case tests

### PR DESCRIPTION
**Pull request by AndreiEres**
Part of https://github.com/paritytech/smoldot/issues/3142

Extended wire protocol V1 tests to match polkadot-sdk coverage: added field ordering violations (duplicates, out-of-order), malformed bytes (invalid discriminants, corrupted proof variant, inflated field count), truncated payloads (empty, partial expiry, partial proof), and cross-compatibility hex vectors for Ed25519 and Secp256k1Ecdsa proofs generated from polkadot-sdk.

### CODE REVIEW

**Review by alexggh — APPROVED** [review]

File: lib/src/network/codec/statement.rs [hunk 1]
```diff
@@ -613,5 +754,35 @@ mod tests {
         assert_eq!(decoded.channel, Some([0xcc; 32]));
 
         assert_eq!(encode_statement(&decoded).unwrap(), bytes);
+
+        // Ed25519 proof: signature=[0xAA; 64], signer=[0xBB; 32], expiry=1000, data=[1,2,3]
+        // Generated by polkadot-sdk's Statement::encode()
+        let ed25519_bytes = hex::decode("0c0001aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaabbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb02e803000000000000080c010203").unwrap();
```
Comment (dmitry-markin) [inline lib/src/network/codec/statement.rs]: nit: Can we split this into lines under 100 characters please?

File: lib/src/network/codec/statement.rs [hunk 2]
```diff
@@ -613,5 +754,35 @@ mod tests {
...
+            Some(Proof::Ed25519 { signature, signer })
+            if signature == [0xAA; 64] && signer == [0xBB; 32]
+        ));
+        assert_eq!(decoded.expiry, 1000);
+        assert_eq!(decoded.data.as_deref(), Some([1, 2, 3].as_slice()));
+        assert_eq!(encode_statement(&decoded).unwrap(), ed25519_bytes);
+
+        // Secp256k1Ecdsa proof: signature=[0x77; 65], signer=[0x88; 33], expiry=500
+        // Generated by polkadot-sdk's Statement::encode()
+        let secp_bytes = hex::decode("080002777777777777777777777777777777777777777777777777777777777777777777777777777777777777777777777777777777777777777777777777777777777788888888888888888888888888888888888888888888888888888888888888888802f401000000000000").unwrap();
```
Comment (dmitry-markin) [inline lib/src/network/codec/statement.rs]: < 100 characters please :)

**Review by dmitry-markin — APPROVED** [review]
